packages/host-portal-app/src/components/remote-view.tsx

```diff
@@ -10,55 +9,69 @@ import React, {
 import { Manifest, View } from '../types';
 
 const loading = Symbol('loading');
-type MicroFrontendState = Record<string, React.ComponentType | typeof loading>;
 
-const views = createContext<
-  [
+type MicroFrontendState = Record<string, React.ComponentType | typeof loading>;
+type GlobalStylesheetSet = Set<string>;
+type RemoteViewsContext = {
+  microFrontends: [
     MicroFrontendState,
     (value: (prevState: MicroFrontendState) => MicroFrontendState) => void,
-  ]
->([{}, () => null]);
-
-async function loadRemoteView(baseUrl: string): Promise<ComponentType | void> {
-  const response = await fetch(`${baseUrl}/package.json`);
-  const manifest = (await response.json()) as Manifest;
-
-  // TODO: check for (and eventually load) global CSS
-
-  // Load microfrontend's local style
-  if (manifest.style) {
-    document.head.insertAdjacentHTML(
-      'beforeend',
-      `<link rel='stylesheet' href='${baseUrl}${manifest.style}' />`,
-    );
-  }
-
-  // Dynamically import ESM entrypoint
-  if (manifest.module) {
-    const { default: LoadedView } = (await import(
-      /* webpackIgnore: true */ `${baseUrl}${manifest.module}`
-    )) as View;
+  ];
+  globalStyles: [
+    GlobalStylesheetSet,
+    (value: (prevState: GlobalStylesheetSet) => GlobalStylesheetSet) => void,
+  ];
+};
 
-    return LoadedView;
-  }
-}
+export const viewContext = createContext<RemoteViewsContext>(
+  {} as RemoteViewsContext,
+);
 
 export const useRemoteView = (baseUrl: string): React.ComponentType | null => {
-  const [state, setState] = useContext(views);
-  const current = state[baseUrl];
+  const {
+    microFrontends: [microFrontendsState, setMicroFrontendsState],
+    globalStyles: [globalStylesState, setGlobalStylesState],
+  } = useContext(viewContext);
+
+  const current = microFrontendsState[baseUrl];
 
   useEffect(() => {
-    if (current !== undefined) {
-      return;
+    if (current === undefined) {
+      void loadRemoteView();
     }
 
-    void loadRemoteView(baseUrl).then((LoadedView) => {
-      LoadedView && setState((old) => ({ ...old, [baseUrl]: LoadedView }));
-    });
-  }, [current, baseUrl, setState]);
+    async function loadRemoteView() {
+      const manifest = await loadRemoteManifest(baseUrl);
+
+      // Load local CSS via manifest "style" field
+      if (manifest.style) {
+        injectRemoteCss(`${baseUrl}${manifest.style}`);
+      }
+
+      // Load global (CDN) CSS via manifest "styleImports" field and make sure it's not been already injected in the page
+      manifest.styleImports?.forEach((url) => {
+        if (!globalStylesState.has(url)) {
+          setGlobalStylesState((globalStyles) => globalStyles.add(url));
+          injectRemoteCss(url);
```

Comment:
As we're not awaiting the CSS loading, this could be simplified by querying the node the stylesheet is being injected into (see next comment)

```typescript
manifest.styleImports?.forEach(injectRemoteCss);
```
```

Reply:
Good idea, this semplifies everything and additionally avoids one reload of uitk (which is loaded by the host in the first place). One downside to this is that it needs to be synced with the way Modular synthetic indices inject global CSS (today it's `<link>`, tomorrow might be [CSS Module Scripts](https://web.dev/css-module-scripts/)), but since we own both it's not a big deal.